ch3/exercise3-11/comma.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+func main() {
+	for i := 1; i < len(os.Args); i++ {
+		fmt.Printf("  %s\n", comma(os.Args[i]))
+	}
+}
+
+func comma(s string) string {
+
+	match, _ := regexp.MatchString("\\.", s)
+	signMatch, _ := regexp.MatchString("[+-]", s)
+	var splitString []string
+	splitString = strings.Split(s, ".")
+	var buf bytes.Buffer
+
+	if (signMatch && len(splitString[0][1:]) <= 3) || len(splitString[0]) <= 3{
+		return s
+	} 
+
+	if match && signMatch {
+		sign := splitString[0][0]
+		buf.WriteByte(sign)
+		buf = insertCommas(buf, splitString[0][1:])
+		buf.WriteString("." + splitString[1])
+		return buf.String()
+
+	} else if match == false && signMatch {
+		sign := splitString[0][0]
+		buf.WriteByte(sign)
+		buf = insertCommas(buf, splitString[0][1:])
+		return buf.String()
+
+	} else if match {
+		buf = insertCommas(buf, splitString[0])
+		buf.WriteString("." + splitString[1])
+		return buf.String()
+	}
```

Comment:
I'd recommend handling 
```
sign := s[0]
buf.WriteByte(sign)
s = s[1:]
```
in a separate conditional beforehand, and keeping this conditional focused on the decimal logic.  It isn't that this is wrong, it is just more difficult to understand because handling of the sign is mixed up with handling of decimals.